### PR TITLE
In Edit mode, User is not able to edit if Repeat(Interval) fields are blank(16)

### PR DESCRIPTION
**Pull request by shariq1501**
Fixed form validation issue on wrong input type for repeat and time field.

### CODE REVIEW

File: src/frontend/src/app/update-modal/update-modal.component.ts [hunk 1]
```diff
@@ -22,23 +22,22 @@ export class UpdateModalComponent implements OnInit, OnChanges {
...
       name: ['', [<any>Validators.required]],
       repeatDay: ['', [<any>Validators.required]],
-      repeat: ['', [<any>Validators.required]],
+      repeat: ['', [<any>Validators.required, Validators.pattern(regExp)]],
       exclusive: [Validators.required],
       process_name: [Validators.required],
       type: [Validators.required],
       day: [Validators.required],
-      time: ['', Validators.required],
+      time: ['', [Validators.required, Validators.pattern(regExp)]],
```
Comment (praveen-garg) [inline src/frontend/src/app/update-modal/update-modal.component.ts:40]: using same regexPattern?

File: src/frontend/src/app/update-modal/update-modal.component.ts [hunk 2]
```diff
@@ -168,9 +161,12 @@ export class UpdateModalComponent implements OnInit, OnChanges {
         if (data.error) {
           console.log('error in response', data.error);
           this.alertService.error(data.error.message);
+        } else {
+          this.alertService.success('Record updated successfully.');
```
Comment (praveen-garg) [inline src/frontend/src/app/update-modal/update-modal.component.ts]: REcord? you mean `Schedule` ? Does not we get a informational message from REST api which we can display to user?

File: src/frontend/src/app/update-modal/update-modal.component.html [hunk 3]
```diff
@@ -10,13 +10,13 @@
         <form [formGroup]="form" (ngSubmit)="f.form.valid && updateSchedule()" #f="ngForm" novalidate>
           <div class="columns">
             <div class="column">
-              <div class="field" [ngClass]="{ 'has-error': f.submitted && !form.controls.name.valid }">
```
Comment (praveen-garg) [inline src/frontend/src/app/update-modal/update-modal.component.html:13]: what is the impact of f.submitted ; we need to make sure; Cancel is not broken

**Review by praveen-garg — CHANGES_REQUESTED** [review]

File: src/frontend/src/app/update-modal/update-modal.component.html (hunk 3 above)
Comment (shariq1501) [reply src/frontend/src/app/update-modal/update-modal.component.html:13]: it was to keep track on whether form is submitted or not but f.submitted was unused here so I have removed that. Cancel in not broken and working fine.

File: src/frontend/src/app/update-modal/update-modal.component.ts (hunk 1 above)
Comment (shariq1501) [reply src/frontend/src/app/update-modal/update-modal.component.ts:40]: yes, this regexPattern is only to verify valid entry in time field and in both case it is `00:00:00`

File: src/frontend/src/app/update-modal/update-modal.component.ts (hunk 2 above)
Comment (shariq1501) [reply src/frontend/src/app/update-modal/update-modal.component.ts]: There is no informational message from api side. We are only getting updated schedule record back in response.

**Comment by MonikaSharma06** [comment]
Approved!

**Pull request merged**